src/components/MAPI/apps/AppList/index.tsx

```diff
@@ -0,0 +1,210 @@
+  }
+};
+
+const AppList: React.FC<{}> = () => {
+  const isAdmin = useSelector(getUserIsAdmin);
+
+  const clientFactory = useHttpClientFactory();
+  const auth = useAuthProvider();
+
+  const appCatalogListClient = useRef(clientFactory());
+  const appCatalogListGetOptions: applicationv1alpha1.IGetAppCatalogListOptions = useMemo(() => {
+    // Admins can see any type of catalogs.
+    if (isAdmin) return {};
+
+    return {
+      labelSelector: {
+        matchingLabels: {
+          [applicationv1alpha1.labelCatalogVisibility]: 'public',
+          [applicationv1alpha1.labelCatalogType]: 'stable',
+        },
+      },
+    };
+  }, [isAdmin]);
+
+  const {
+    data: appCatalogList,
+    error: appCatalogListError,
+    isValidating: appCatalogListIsValidating,
+  } = useSWR<applicationv1alpha1.IAppCatalogList, GenericResponse>(
+    applicationv1alpha1.getAppCatalogListKey(appCatalogListGetOptions),
+    () =>
+      applicationv1alpha1.getAppCatalogList(
+        appCatalogListClient.current,
+        auth,
+        appCatalogListGetOptions
+      )
+  );
+  const prevAppCatalogList = usePrevious(appCatalogList);
+
+  useEffect(() => {
+    if (appCatalogListError) {
+      const errorMessage = extractErrorMessage(appCatalogListError);
+
+      new FlashMessage(
+        'There was a problem loading app catalogs.',
+        messageType.ERROR,
+        messageTTL.FOREVER,
+        errorMessage
+      );
+    }
+  }, [appCatalogListError]);
+
+  const appCatalogListIsLoading =
+    typeof appCatalogList === 'undefined' && appCatalogListIsValidating;
+
+  const [searchQuery, setSearchQuery] = useState('');
+  const debouncedSearchQuery = useDebounce(
+    searchQuery,
+    SEARCH_THROTTLE_RATE_MS
+  );
+
+  const [selectedCatalogs, dispatchSelectedCatalogs] = useReducer(
+    selectedCatalogsReducer,
+    {}
+  );
+
+  useLayoutEffect(() => {
+    // Only execute this after the initial catalog load.
+    if (
+      Object.keys(selectedCatalogs).length > 0 ||
+      typeof prevAppCatalogList !== 'undefined' ||
+      typeof appCatalogList === 'undefined'
+    ) {
+      return;
+    }
+
+    // Pre-select all catalogs except `helm-stable`.
+    for (const catalog of appCatalogList.items) {
+      if (catalog.metadata.name === 'helm-stable') continue;
+
+      dispatchSelectedCatalogs({
+        type: 'selectCatalog',
+        name: catalog.metadata.name,
+      });
+    }
+
+    // We don't need to run this again if the selected catalogs change.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [appCatalogList?.items, prevAppCatalogList]);
+
+  const {
+    data: appCatalogIndexAppList,
+    isValidating: appCatalogIndexAppListIsValidating,
+  } = useSWR<IAppCatalogIndexAppList, GenericResponse>(
+    getAppCatalogsIndexAppListKey(appCatalogList?.items),
+    // TODO(axbarsan): Find a more elegant solution for passing `fetch` here.
+    () => getAppCatalogsIndexAppList(fetch, auth, appCatalogList!.items)
```

Comment:
Had to use `fetch` instead of our HTTP client, since our client doesn't support enforcing CORS from the client-side, which we need for fetching catalogs from GitHub.